platform/warm.go

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package platform
+
+import (
+	"context"
+	"errors"
+	v08 "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func NewWarmAction(reader ctrl.Reader) Action {
+	return &warmAction{
+		reader: reader,
+	}
+}
+
+type warmAction struct {
+	baseAction
+	reader ctrl.Reader
+}
+
+func (action *warmAction) Name() string {
+	return "warm"
+}
+
+func (action *warmAction) CanHandle(platform *v08.KogitoServerlessPlatform) bool {
+	return platform.Status.Phase == v08.PlatformPhaseWarming
+}
+
+func (action *warmAction) Handle(ctx context.Context, platform *v08.KogitoServerlessPlatform) (*v08.KogitoServerlessPlatform, error) {
```

Comment:
what the "warm" term means for Kaniko?

Reply:
The "warm" term is intended for Kaniko as the action to pull a base image used into the build process before the first container image build will start.

Reply:
ok, thanks.